onnxruntime/core/optimizer/gather_internal_replacement.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// gather_internal_replacement.cc
+
+#include "core/optimizer/gather_internal_replacement.h"
+
+#include "core/common/logging/logging.h"
+#include "core/optimizer/rewrite_rule.h"
+#include "core/optimizer/utils.h"
+#include "core/graph/graph.h"
+#include "core/graph/graph_utils.h"
+
+namespace onnxruntime {
+
+namespace {
+
+const char* GatherGradOpType = "GatherGrad";
+const char* GatherName = "Gather";
+
+}
+
+bool GatherInternalReplacement::SatisfyCondition(const Graph&, const Node& gather_internal_node, const logging::Logger&) const
+{
+    // If the gather_internal_node is not connected to a GatherGrad node, then the extra
+    // precomputed GatherInternal outputs are not needed. And the node can be replaced
+    // by a simple Gather node.
+    for (auto output_node = gather_internal_node.OutputNodesBegin();
+         output_node != gather_internal_node.OutputNodesEnd();
+         ++output_node)
+    {
+        if (output_node->OpType() == GatherGradOpType)
+        {
```

Comment:
clang format

consider installing "Clang-Format" vscode extension.

Reply:
Thanks. I formatted the changes accordingly.